### PR TITLE
Adaptation to MvStore, multithreaded consumers, retry scheduling

### PR DESCRIPTION
**Pull request by jamieb22**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -33,16 +33,15 @@ the consumption of the message by another thread.
      * This example shows how to setup the queue, how to consume it
      * and how to send a message.
      */
-    public class RichBDBQueueExample {
+    public class DBQueueExample {
         
-        private static final Logger LOG = LoggerFactory.getLogger(RichBDBQueueExample.class);
+        private static final Logger LOG = LoggerFactory.getLogger(BDBQueueExample.class);
```
Comment (magro) [inline README.markdown]: This should be `LoggerFactory.getLogger(DBQueueExample.class);`

File: core/README.markdown [hunk 2]
```diff
@@ -33,16 +33,15 @@ the consumption of the message by another thread.
      * This example shows how to setup the queue, how to consume it
      * and how to send a message.
      */
-    public class RichBDBQueueExample {
+    public class DBQueueExample {
         
-        private static final Logger LOG = LoggerFactory.getLogger(RichBDBQueueExample.class);
+        private static final Logger LOG = LoggerFactory.getLogger(BDBQueueExample.class);
```
Comment (magro) [inline core/README.markdown]: Again DBQueueExample here...

File: core/src/main/java/de/javakaffee/simplequeue/Consumer.java [hunk 3]
```diff
@@ -11,15 +13,16 @@
...
  */
-package de.javakaffee.simplequeue;
 
 /**
- * A queue consumer, ca be used for {@link RichBDBQueue#consume(Consumer)}.
- * 
- * Created on Jun 27, 2011
+ *
+ * Fast queue implementation on top of MVStore. This class is thread-safe.
+ * Adaptation of Martin Grotzeke's work
```
Comment (magro) [inline core/src/main/java/de/javakaffee/simplequeue/Consumer.java]: It's "Grotzke", not "Grotzeke" :-)

File: core/src/main/java/de/javakaffee/simplequeue/DBQueue.java [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+ * Created on Jun 22, 2011
+ *
+ * @author Martin Grotzke (initial creation)
+ * @author Stimulus Software (adaptation to MVStore, added multithreaded consumer support & retry delay option)
+ */
+public class DBQueue<T> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DBQueue.class);
+	public final static long SECOND_MILLIS = 1000;
+    private final ObjectMapper _mapper;
```
Comment (magro) [inline core/src/main/java/de/javakaffee/simplequeue/DBQueue.java]: Cleanup indentation?

**Comment by magro** [comment]
Thanks for the pull request! I added some comments.

A general one: I'd probably not force users to subclass RetryQueueItem, but keep that state in a Map (perhaps IdentityHashMap) keyed by the queue items. What do you think?

**Comment by jamieb22** [comment]
Martin

I agree that its not very nice to force users to subclass 
RetryQueueItem, but I needed to persistent the retry attempt & last 
retry date. If you store the state in a map, presumably, the retry state 
would be lost on restart? Other alternatives would be to use annotations 
or injection.

Jamie

**Comment by magro** [comment]
Would it be an issue if the retry state
would be lost on restart?
Am 27.08.2014 18:02 schrieb "jamieb22" notifications@github.com:

**Comment by jamieb22** [comment]
For me, it is. Look, as it stands at the moment, subclasses 
RetryQueueItem is optional. I know that the instanceof check is an 
irritation, but there are times when one must accept a suboptimal design.

**Comment by magro** [comment]
Ok, then I'd suggest to change this to an interface and add an
AbstractRetryQueueItem.
Am 27.08.2014 21:10 schrieb "jamieb22" notifications@github.com:

**Comment by jamieb22** [comment]
I am happy with that. Since you're the original author, you're the boss 
anyhow!

**Comment by magro** [comment]
Fine :-) When you update the pull request I'll merge it.
Am 27.08.2014 21:14 schrieb "jamieb22" notifications@github.com:

**Pull request closed without merging**